ledger-core/v2/virtual/execute/update_pending_counters_test.go

```diff
@@ -225,8 +225,8 @@ func Test_SlotMachine_Increment_Pending_Counters(t *testing.T) {
 		QueryRoleMock.Return([]reference.Global{gen.Reference()}, nil)
 	pulses := pulse.NewStorageMem()
 	messageSender := messagesender.NewDefaultService(publisherMock, jetCoordinatorMock, pulses)
-	messageSenderAdapter := messagesenderadapter.CreateMessageSendService(ctx, messageSender)
-	slotMachine.AddDependency(messageSenderAdapter)
+	var messageSenderAdapter messagesenderadapter.MessageSender = messagesenderadapter.CreateMessageSendService(ctx, messageSender)
```

Comment:
use, instead:
```suggestion
	messageSenderAdapter := messagesenderadapter.CreateMessageSendService(ctx, messageSender)
```

Reply:
this is is intentional. CreateMessageSendService returns exact implementation. We need interface.

Reply:
Yes, it is very important use interface here